### PR TITLE
Velychkomaryna/feature/signup child integration

### PR DESCRIPTION
**Pull request by maryna-velychko**

### CODE REVIEW

File: src/app/(auth)/signup/components/SignUpChildren.js [hunk 1]
```diff
@@ -4,23 +4,46 @@ import {
...
+import { useState } from "react"
 import * as Yup from "yup";
 import { SignUpContainer } from "./SignUpAccount";
+import { signupChildren, USER_ID_KEY } from "@/app/api.mjs";
 
 export default function SignUpChildren({ goToNextStep }) {
   const { data: schools } = useQuery("schools", getSchools);
+  const [stayOnPage, setStayOnPage] = useState(false);
+  const [childNumber, setChildNumber] = useState(1);
+  const userId = localStorage.getItem(USER_ID_KEY);
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/components/SignUpChildren.js:26]: Why we do not utilize useEffect here and do utilize on another page?

File: src/app/(auth)/signup/components/SignUpChildren.js [hunk 2]
```diff
@@ -4,23 +4,46 @@ import {
...
+import { useMutation, useQuery } from "react-query";
+import { useState } from "react"
 import * as Yup from "yup";
 import { SignUpContainer } from "./SignUpAccount";
+import { signupChildren, USER_ID_KEY } from "@/app/api.mjs";
 
 export default function SignUpChildren({ goToNextStep }) {
   const { data: schools } = useQuery("schools", getSchools);
+  const [stayOnPage, setStayOnPage] = useState(false);
+  const [childNumber, setChildNumber] = useState(1);
```
[the remark this answers is not in the document]
Comment (maryna-velychko) [reply src/app/(auth)/signup/components/SignUpChildren.js:25]: However, users cannot return to this page(except change currentStep manually in local storage)

File: src/app/(auth)/signup/components/SignUpChildren.js (hunk 1 above)
Comment (maryna-velychko) [reply src/app/(auth)/signup/components/SignUpChildren.js:26]: I didn't get the error on this page that local storage is not defined. Problem with next.js and SSR [explain](https://sentry.io/answers/referenceerror-localstorage-is-not-defined-in-next-js/). I think, it is a difference between page and a component rendering:

File: src/app/(auth)/signup/components/SignUpChildren.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (shuryhin-oleksandr) [reply src/app/(auth)/signup/components/SignUpChildren.js:25]: Just close tab and open again

File: src/app/(auth)/signup/components/SignUpChildren.js (hunk 1 above)
Comment (shuryhin-oleksandr) [reply src/app/(auth)/signup/components/SignUpChildren.js:26]: Why then do not mark a previous case a client side to make it consistent?

Comment (maryna-velychko) [reply src/app/(auth)/signup/components/SignUpChildren.js:25]: I have am error: `localStorage is not defined` despite using the 'use client' directive. This typically happens when localStorage is accessed before the component is mounted on the client side.

File: src/app/api.mjs [hunk 3]
```diff
@@ -209,3 +210,9 @@ export async function signupDetails(userId, data) {
   const response = await client.put(url, data);
   return response.data;
 }
+
+export async function signupChildren(userId, data) {
+  const url = `${SIGNUP_CHILDREN_PATH}?userId=${userId}`;
+  const response = await client.post(url, data);
+  return response.data;
+}
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:218]: PEP-8

File: src/app/(auth)/signup/components/SignUpChildren.js [hunk 4]
```diff
@@ -4,23 +4,44 @@ import {
...
-import { useQuery } from "react-query";
+import { useMutation, useQuery } from "react-query";
+import { useState } from "react"
 import * as Yup from "yup";
 import { SignUpContainer } from "./SignUpAccount";
+import { signupChildren, USER_ID_KEY } from "@/app/api.mjs";
 
 export default function SignUpChildren({ goToNextStep }) {
   const { data: schools } = useQuery("schools", getSchools);
+  const [stayOnPage, setStayOnPage] = useState(false);
```
Comment (shuryhin-oleksandr) [inline src/app/(auth)/signup/components/SignUpChildren.js:24]: Is this a best way to handle page transitions?

**Pull request merged**